client/src/Components/Layouts/SingleCaptains/Popups/deletePopup.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+import { Modal, notification } from "antd";
+
+class Popup extends Component {
+  state = {
+    visible: this.props.visible,
+    id: this.props.id
+  };
+
+  openNotificationWithIcon = (type, message) => {
+    notification[type]({
+      message: message,
+      duration: 1.5
+    });
+  };
+
+  handleOk = e => {
+    fetch(`/api/v1/deleteOrder/${this.props.id}`, { method: "DELETE" })
+      .then(res => res.json())
+      .then(res => {
+        const { error } = res;
+        if (error) {
+          this.openNotificationWithIcon("error", error);
+        } else {
+          this.props.updateState(this.state.id);
+          this.openNotificationWithIcon("success", "Delete Done");
+        }
+        this.props.visibleFun("singleCaptain", "deleteVisibility")(e);
+      })
+      .catch(() => {
+        this.openNotificationWithIcon('warning', 'Something error please try again');
+        this.props.visibleFun("singleCaptain", "deleteVisibility")(e);
+      });
+  };
+
+  handleCancel = e => {
+    this.props.visibleFun("singleCaptain", "deleteVisibility")(e);
+  };
+  componentWillReceiveProps(props) {
+    const { visible, id } = props;
+    this.setState({ visible, id });
+  }
+
+  render() {
+    return (
+      <Modal
+        title="حذف الطلب"
+        visible={this.state.visible}
```

Comment:
@naremanhilles Also here, it would probably be simpler if you used this.props.visible directly, instead of copying props.visible into state.visible on mount and willReceiveProps.